### ISSUE TITLE
Selection formatting mode (VS Code v1.49 feature support)

### ISSUE THREAD

**Issue by GriceTurrble** [psf/black#1706]
**Is your feature request related to a problem? Please describe.** Visual Studio Code v1.49 offers a "modifications only" formatting mode, in which it will run the selected formatter only on modified lines of the file on save: https://code.visualstudio.com/updates/v1_49#_only-format-modified-text

However, Black cannot currently support selection formatting, so it is not possible to use this feature.

**Describe the solution you'd like** Add support for Black to format selected ranges of code within a file in support of VS Code's feature.

**Comment by ichard26** [comment]
Duplicate of #830 or #1352 depending on what kind of solution and workflow you want. #830 wants to add a `--lines` CLI option that tells Black which part of the code it is allowed to touch. While #1352 just wants Black to able to process indented code (which is probably a section of a file) and not error out. Of course #830 depends on #1352, but not the otherway around.

Thanks for the suggestion anyhow!
